b2share/modules/records/triggers.py

```diff
@@ -60,12 +70,18 @@ def set_record_schema(record, **kwargs):
     record['$schema'] = community_schema_json_schema_link(schema)
 
 
+def set_record_owner(record, **kwargs):
+    if not current_user.is_authenticated:
```

Comment:
I don't disagree, but I want to make sure in this function that user.id exists and is valid. More of a double check. If the permission check comes first, it's even better. I will change this to abort(401) as you said.